internal/consensus/state.go

```diff
@@ -2266,7 +2266,13 @@ func (cs *State) updatePrivValidatorPubKey() error {
 		timeout = cs.config.TimeoutPrevote
 	}
 
-	// set a hard timeout for 2 seconds. This helps in avoiding blocking of the remote signer connection
+	// no GetPubKey retry beyond the proposal/voting
+	if cs.Step >= cstypes.RoundStepPrecommit {
+		timeout = 0
```

Comment:
in this case we create a context on line 2276 that is immediately canceled, which strikes me as not the intentional behavior. (e.g. we should just return an error here)

Reply:
Thanks for the catch, right, I should return an error here directly.

Reply:
I do not believe this is an error case -- the intention is to only loop once in the get public key function.

Reply:
I updated the code with the signerType check. Only RetrySignerClient will apply this rule.

Reply:
Thanks, @tychoish,  looks like the gRPC Signer Client doesn't have the retry logic, correct?

Reply:
not sure I can add `Type()` to the interface `PrivValidator`? If yes, the code will be neater.

Reply:
@alexanderbez if we hit this condition then it's always going to be an error because the context will canceled, so I think Jay's plan seems reasonable.